### PR TITLE
fix MGDCTRS-1786: MongoDB sink fails with Prematurely reached end of stream

### PR DESCRIPTION
**Pull request by valdar**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by rinaldodev — APPROVED** [review]

File: cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            MongoClientSettings settings = MongoClientSettings.builder()
+                    .applyToSslSettings(builder -> {
+                        builder.enabled(ssl);
+                        if (sslAllowInvalidCertificatesAndHostName) {
+                            builder.invalidHostNameAllowed(true);
+                            SSLContext sc = null;
+                            try {
+                                sc = SSLContext.getInstance("SSL");
+                            } catch (NoSuchAlgorithmException e) {
+                                e.printStackTrace();
```
Comment (lburgazzoli) [inline cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java:67]: should perhaps re-throw a runtime exception ?

File: cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                            }
+                            try {
+                                sc.init(null, trustAllCerts, new java.security.SecureRandom());
+                            } catch (KeyManagementException e) {
+                                e.printStackTrace();
```
Comment (lburgazzoli) [inline cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java:72]: should perhaps re-throw a runtime exception ?

File: cos-fleet-catalog-kamelets/mongodb/src/main/resources/kamelets/cos-mongodb-sink.kamelet.yaml [hunk 3]
```diff
@@ -55,6 +55,20 @@ spec:
...
         x-descriptors:
           - urn:camel:group:credentials
+      ssl:
+        title: Enable ssl for mongodb connection
+        description: whether to enable ssl connection to mongodb
+        type: boolean
+        default: true
+        x-descriptors:
+          - 'urn:alm:descriptor:com.tectonic.ui:checkbox'
+      sslAllowInvalidCertificatesAndHostName:
```
[the remark this answers is not in the document]
Comment (valdar) [reply cos-fleet-catalog-kamelets/mongodb/src/main/resources/kamelets/cos-mongodb-sink.kamelet.yaml:65]: something like `sslDisableAllValidations` ?

File: cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java (hunk 2 above)
Comment (valdar) [reply cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java]: done

File: cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java (hunk 1 above)
Comment (valdar) [reply cos-fleet-catalog-kamelets/mongodb/src/main/java/org/bf2/cos/connector/camel/mongodb/SslAwareMongoClient.java]: done

**Review by lburgazzoli — APPROVED** [review]

**Pull request merged**